src/src/PluginStructs/P104_data_struct.h

```diff
@@ -0,0 +1,440 @@
+  char     buffer[P104_SETTINGS_BUFFER]; // Max. acceptable value would be ~1020
+};
+
+struct P104_data_struct : public PluginTaskData_base {
+  P104_data_struct(MD_MAX72XX::moduleType_t _mod,
+                   taskIndex_t              _taskIndex,
+                   int8_t                   _cs_pin,
+                   uint8_t                  _modules);
+
+  bool   begin();
+  void   loadSettings();
+  bool   saveSettings();
+  bool   webform_load(struct EventStruct *event);
+  bool   webform_save(struct EventStruct *event);
+  String getError() {
+    return error;
+  }
+
+  void configureZones();
+
+  void setZones(uint16_t _zones) {
+    expectedZones = _zones;
+  }
+
+  bool handlePluginWrite(taskIndex_t   taskIndex,
+                         const String& string);
+  bool handlePluginOncePerSecond(struct EventStruct *event);
+  void checkRepeatTimer(uint8_t z);
+  void updateZone(uint8_t                 zone,
+                  const P104_zone_struct& zstruct);
+
+  MD_Parola *P = nullptr;
+
+  bool logAllText = false;
+
+private:
+
+  # ifdef P104_USE_BAR_GRAPH
+  MD_MAX72XX *pM = nullptr;
+  void displayBarGraph(uint8_t                 zone,
+                       const P104_zone_struct& zstruct,
+                       const String          & graph);
+  void modulesOnOff(uint8_t                    start,
+                    uint8_t                    end,
+                    MD_MAX72XX::controlValue_t on_off);
+  void drawOneBarGraph(uint16_t lower,
+                       uint16_t upper,
+                       int16_t  pixBottom,
+                       int16_t  pixTop,
+                       uint16_t zeroPoint,
+                       uint8_t  barWidth,
+                       uint8_t  barType,
+                       uint8_t  row);
+  # endif // ifdef P104_USE_BAR_GRAPH
+
+  void   displayOneZoneText(uint8_t                 currentZone,
+                            const P104_zone_struct& idx,
+                            const String          & text);
+  String enquoteString(const String& input);
+
+  String error;
+
+  // Stored settings
+  tP104_StoredSettings StoredSettings;
+
+  std::vector<P104_zone_struct>zones;
+  String                       sZoneBuffers[P104_MAX_ZONES];
+  String                       sZoneInitial[P104_MAX_ZONES];
+
+  MD_MAX72XX::moduleType_t mod;
+  uint16_t                 numDevices = 0;
+  taskIndex_t              taskIndex;
+  int8_t                   cs_pin;
+  uint8_t                  modules          = 1u;
+  int8_t                   expectedZones    = -1;
+  int8_t                   previousZones    = -1;
+  uint8_t                  zoneOrder        = 0;
+  bool                     initialized      = false;
+  bool                     zonesInitialized = false;
+  bool                     flasher          = false; // seconds passing flasher
+
+  // time/date stuff
+  char szTimeL[P104_MAX_MESG];                       // dd-mm-yy mm:ss\0
+  char szTimeH[P104_MAX_MESG];
+};
```

Comment:
The `sizeof(P104_data_struct)` is probably quite large and thus it is hard to create it as it needs to be a single large block.
`tP104_StoredSettings` does need a stack-allocated array buffer and is also just a "stack allocated" member of this class (not a pointer)
There is also a number of arrays of type String and some char arrays for time/date stuff.
I think changing `StoredSettings` into a pointer and create it in the constructor will make things a bit easier to allocate as you don't need to have such a large continuous free block.

N.B. I called it "stack allocated", by lack of a better description.
It isn't allocated inside the object via `new`, so the `sizeof()` that member adds up to the `sizeof()` of the data struct object.
The `P104_data_struct` is allocated on the heap using `new`, so you don't have to worry about the small 4k stack we have. But the objects you need to allocate are quite big on their own, so they may cause issues on a fragmented heap.